### PR TITLE
Improve the block HTML identification

### PR DESCRIPTION
**Pull request by srawlins**
Better CommonMark [HTML Blocks](http://spec.commonmark.org/0.24/#html-blocks). This makes us compliant with 29 more CommonMark 0.24 tests.

All of the existing tests pass.

### CODE REVIEW

File: lib/src/block_parser.dart [hunk 1]
```diff
@@ -65,9 +61,17 @@ class BlockParser {
...
 
   /// The collection of built-in block parsers.
-  final List<BlockSyntax> standardBlockSyntaxes = const [
+  final List<BlockSyntax> standardBlockSyntaxes = [
     const EmptyBlockSyntax(),
-    const BlockHtmlSyntax(),
+    const BlockTagBlockHtmlSyntax(),
+    new LongBlockHtmlSyntax(r'^ {0,3}<pre(?:\s|>|$)', '</pre>'),
+    new LongBlockHtmlSyntax(r'^ {0,3}<script(?:\s|>|$)', '</script>'),
+    new LongBlockHtmlSyntax(r'^ {0,3}<style(?:\s|>|$)', '</style>'),
```
Comment (munificent) [inline lib/src/block_parser.dart]: How come these don't use BlockTagBlockHtmlSyntax? Can you add a comment explaining?

File: lib/src/block_parser.dart [hunk 2]
```diff
@@ -413,6 +424,40 @@ class BlockHtmlSyntax extends BlockSyntax {
   }
 }
 
+class OtherTagBlockHtmlSyntax extends BlockTagBlockHtmlSyntax {
+  bool get canEndBlock => false;
+
+  // Really hacky way to detect "other" HTML.
+  RegExp get pattern => new RegExp(r'^ {0,3}</?\w+(?:>|\s+[^>]*>)\s*$');
```
Comment (munificent) [inline lib/src/block_parser.dart]: This RegExp hurts my mind. How about a comment explaining it?

Comment (srawlins) [inline lib/src/block_parser.dart]: Done.

File: lib/src/block_parser.dart (hunk 1 above)
Comment (srawlins) [inline lib/src/block_parser.dart]: Sorry, added a comment to LongBlockHtmlSyntax:

> A BlockHtmlSyntax that has a specific [endPattern].
>
> In practice this means that the syntax dominates; it is allowed to eat
> many lines, including blank lines, before matching its [endPattern].

**Comment by munificent** [comment]
👍

**Pull request merged**